### PR TITLE
Send the data to the server-end

### LINKED ISSUE #5: Send the data to the server-end
- [x] setup the development environment
- [x] replace the current save CSV to send out the request
- [x] save logs for debugging purpose

### LINKED ISSUE #15: Post request problem
**Describe** 
When I do the post request function in the Android code, it shows some problems I cannot fix. It seems that the URL is not good for doing post request, because I am using localhost to do the server website. 
1. When I use ‘http://localhost/api/v1/stress’, the app cannot work well, when the function is started, and the app will be shut down directly.  I try to find the error from the log, but it shows nothing can help, so I try to make a new folder to run this function, and I got the log of this error: ‘NetworkSecurityPolicy.getInstance().isCleartextTrafficPermitted()’, and I search this in the google, I got the solution (https://stackoverflow.com/questions/45940861/android-8-cleartext-http-traffic-not-permitted) and it works for me.
image
2. However, when I merge this code into our project, it works better than before, but it still has errors,  'java.net.ConnectException: Failed to connect to localhost/127.0.0.1:80', also, I try to find a solution on google: (https://stackoverflow.com/questions/36811202/java-net-connectexception-fail-to-connect-to-localhost-127-0-0-1port-8080-co), it seems that I should not use localhost URL, and I also tried to change the localhost to the ‘127.0.0.1’ or delete the 'http://', it still not work.
3. I have tested another URL(https://reqres.in/api/users?page=2), it works well and I send post request successfully, I am not sure what the real reason for this bug. 
image

**Reference**
https://developer.android.com/training/volley/request
https://stackoverflow.com/questions/45940861/android-8-cleartext-http-traffic-not-permitted
https://stackoverflow.com/questions/36811202/java-net-connectexception-fail-to-connect-to-localhost-127-0-0-1port-8080-co

### PR DESCRIPTION
**Pull request by ggao007**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: #5 

## What is the new behavior?

-Fix the the issue problem #15
- Send the data to the end-point from the app

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: client/app/src/main/res/xml/network_security_config.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <domain-config cleartextTrafficPermitted="true">
+        <domain includeSubdomains="true">172.23.49.75</domain>
+    </domain-config>
```
Comment (SiyuQian) [inline client/app/src/main/res/xml/network_security_config.xml]: I am thinking that is it possible that we allow all the IP addresses from 172.xx.xx.xx? So in the future test, we don't need to modify the code every time we reconnect to a wifi connection.

**Review by SiyuQian — CHANGES_REQUESTED** [review]

File: xml/network_security_config.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <domain-config cleartextTrafficPermitted="true">
+        <domain includeSubdomains="true">172.23.44.176</domain>
+    </domain-config>
+</network-security-config>
```
Comment (SiyuQian) [inline xml/network_security_config.xml:6]: Is it a duplication with the ```res/xml/network_security_config.xml``` file? I don't know ... If both files are needed please ignore me.

**Comment by SiyuQian** [comment]
@ggao007 I think we need to send the data every > 10 seconds.  https://github.com/neuropsychology/NeuroKit#interval-related

File: client/app/src/main/res/xml/network_security_config.xml (hunk 1 above)
Comment (ggao007) [reply client/app/src/main/res/xml/network_security_config.xml]: I think we will run this experiment in the same room of the university, and all participant will in the same room, so I don't think we need to change IP for each time.

**Review by SiyuQian — APPROVED** [review]

**Pull request merged**